include/boost/geometry/algorithms/detail/overlay/traversal.hpp

```diff
@@ -589,19 +589,22 @@ public :
         {
             // Check counts: in some cases interior rings might be generated with
             // polygons on both sides. For dissolve it can be anything.
-            return 0;
+
+            // If this forms a spike, going to/from the cluster point in the same
+            // (opposite) direction, it can still be used.
+            return cinfo.spike_count > 0 ? 1 : 0;
         }
 
         bool const to_start = ranked_point.turn_index == start_turn_index;
         bool const to_start_index = ranked_point.operation_index == start_op_index;
 
         bool const next_in_same_cluster
-                = cluster_indices.count(op.enriched.get_next_turn_index()) > 0;
+                = cinfo.turn_indices.count(op.enriched.get_next_turn_index()) > 0;
 
-        return to_start && to_start_index ? 4
-            : to_start ? 3
-            : next_in_same_cluster ? 1
-            : 2
+        return to_start && to_start_index ? 5
+            : to_start ? 4
+            : next_in_same_cluster ? 2
+            : 3
```

Comment:
The values just define the ordering. There is now 1 extra, lowest ordering: spike out

Reply:
Is there a comment that explain this orderings?

Reply:
✅ the comment was a bit above, but I didn't update it. Thanks. 
Comment is moved to function description, and a short comment here.